lib/logstash/outputs/graphite.rb

```diff
@@ -145,6 +140,69 @@ def receive(event)
         retry if @resend_on_failure
       end
     end
-
   end # def receive
+
+  private
+
+  def messages_from_event_fields(event, include_metrics, exclude_metrics)
+    timestamp = event_timestamp(event)
+    @logger.debug? && @logger.debug("got metrics event", :metrics => event.to_hash)
+    event.to_hash.flat_map do |metric,value|
```

Comment:
@talevy I have added the requested spec in my last push. Any other requests for improvement here?